### PR TITLE
FIX: Add covariance module to sklearnex init

### PR DESCRIPTION
**Pull request by olegkkruglov**
### Description

* Add covariance module to sklearnex `__init__.py` which allows to use `covariance` submodule after `import *`

The following code will work only after this change

```python
from sklearnex import *
c = covariance.IncrementalEmpiricalCovariance()
```

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.  
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Comment by samir-nasibli** [comment]
@olegkkruglov Why this wasn't before catch? I think make sense add test suit for the import

**Comment by olegkkruglov** [comment]
> @olegkkruglov Why this wasn't before catch?

I guess because we don't use import * in our tests

**Comment by samir-nasibli** [comment]
@olegkkruglov we have this import in the tests https://github.com/intel/scikit-learn-intelex/blob/f094bc40fd8194cbe652dcb17619baf4dc0a87f4/sklearnex/covariance/tests/test_incremental_covariance.py#L222
How this work then?

**Comment by olegkkruglov** [comment]
> @olegkkruglov I think make sense add test suit for the import



> @olegkkruglov we have this import in the tests
> 
> https://github.com/intel/scikit-learn-intelex/blob/f094bc40fd8194cbe652dcb17619baf4dc0a87f4/sklearnex/covariance/tests/test_incremental_covariance.py#L222
> 
> 
> How this work then?

it is not `import *`

for example, the following code will work only after my change
`
from sklearnex import *
c = covariance.EmpiricalCovariance()
`

**Review by samir-nasibli — APPROVED** [review]
CI fails are not related with the changes. But I would suggest to run internal CI once again

File: sklearnex/__init__.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "basic_statistics",
     "cluster",
     "config_context",
+    "covariance",
```
Comment (icfaust) [inline sklearnex/__init__.py:35]: @olegkkruglov I believe your example in the description is incorrect. It should be ```IncrementalEmpiricalCovariance```. Importing ```EmpericalCovariance``` can only be done from the sklearnex.preview package. @samir-nasibli This would have been caught by https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/tests/test_patching.py#L289 but it only looks for matching estimators, which is not yet the case because EmpiricalCovariance is still in preview and IncrementalEmpericalCovariance is a sklearnex-unique estimator.

Comment (samir-nasibli) [reply sklearnex/__init__.py:35]: > @olegkkruglov I believe your example in the description is incorrect. It should be `IncrementalEmpiricalCovariance`. Importing `EmpericalCovariance` can only be done from the sklearnex.preview package. @samir-nasibli This would have been caught by https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/tests/test_patching.py#L289 but it only looks for matching estimators, which is not yet the case because EmpiricalCovariance is still in preview and IncrementalEmpericalCovariance is a sklearnex-unique estimator.

@icfaust thank you! I was confused by the description. Let's cover this with a test

Comment (olegkkruglov) [reply sklearnex/__init__.py:35]: i think the test is not really necessary here. after moving EmpiricalCovariance out of preview this will be covered

Comment (samir-nasibli) [reply sklearnex/__init__.py:35]: Discussed with @icfaust 
We acan adapt this test 
https://github.com/icfaust/scikit-learn-intelex/blob/dev/centralize_finite_assert/sklearnex/tests/test_common.py#L101
if he were to load all_estimators and then cross-reference to sklearnex.__all__
But this suggestion the scope for another PR.

Please @olegkkruglov make sure to create a ticket for this.

**Review by samir-nasibli — APPROVED** [review]
Thank you @olegkkruglov for the finding.
